src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -1835,21 +1854,21 @@ void RadioSpinel<InterfaceType, ProcessContextType>::HandleTransmitDone(uint32_t
     aBuffer += unpacked;
     aLength -= static_cast<uint16_t>(unpacked);
 
-    unpacked = spinel_datatype_unpack(aBuffer, aLength, SPINEL_DATATYPE_BOOL_S, &framePending);
-    VerifyOrExit(unpacked > 0, error = OT_ERROR_PARSE);
-
-    aBuffer += unpacked;
-    aLength -= static_cast<uint16_t>(unpacked);
-
     if (status == SPINEL_STATUS_OK)
     {
+        unpacked = spinel_datatype_unpack(aBuffer, aLength, SPINEL_DATATYPE_BOOL_S, &framePending);
```

Comment:
This change fixes a bug where a spinel error message gets masked by OT_ERROR_PARSE message because the error message can't be further parsed.